pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/geospatial/BaseH3IndexCreator.java

```diff
@@ -108,10 +114,19 @@ public Geometry deserialize(byte[] bytes) {
   }
 
   @Override
-  public void add(Geometry geometry)
+  public void add(@Nullable Geometry geometry)
       throws IOException {
-    Preconditions.checkState(geometry instanceof Point, "H3 index can only be applied to Point, got: %s",
-        geometry.getGeometryType());
+    if (geometry == null || !(geometry instanceof Point)) {
+      if (_tableNameWithType != null) {
+        String metricKeyName =
+            _tableNameWithType + "-" + H3IndexType.INDEX_DISPLAY_NAME.toUpperCase(Locale.US) + "-indexingError";
+        ServerMetrics.get().addMeteredTableValue(metricKeyName, ServerMeter.INDEXING_FAILURES, 1);
+      } else {
```

Comment:
Can `_tableNameWithType` ever be `null`?

Reply:
I've mostly added this due to cases where tableConfig might be null, etc:

```
      if (_tableConfig != null) {
        builder.withTableNameWithType(_tableConfig.getTableName());
      }
```